server/resp/src/main/java/org/infinispan/server/resp/commands/generic/TOUCH.java

```diff
@@ -0,0 +1,40 @@
+package org.infinispan.server.resp.commands.generic;
+
+import io.netty.channel.ChannelHandlerContext;
+import org.infinispan.commons.util.concurrent.CompletableFutures;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+/**
+ * Alters the last access time of a key(s). A key is ignored if it does not exist.
+ * Integer reply: The number of keys that were touched.
+ *
+ * @link https://redis.io/commands/touch/
+ * @since 15.0
+ */
+public class TOUCH extends RespCommand implements Resp3Command {
+   public TOUCH() {
+      super(-2, 1, -1, 1);
+   }
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+                                                      ChannelHandlerContext ctx,
+                                                      List<byte[]> arguments) {
+
+      CompletableFuture<Long> totalTouchCount = CompletableFutures.sequence(
+            arguments.stream()
+                  .map(key -> (CompletableFuture<Boolean>) handler.cache().touch(key, true))
```

Comment:
Do we want to touch even if it is max idle expired?

Reply:
yes, I think so

Reply:
The default behavior should be not to touch a key if it is already max expired. The only reason it is an option is for when a key is read on a different node before when it wasn't expired.